cal/cal/static/js/app.js

```diff
@@ -111,29 +111,41 @@ analyticsApp.component('tags', {
 function TagsDetailCtrl($scope, $http) {
   var tagUrl = '/v1/tags/' + this.tagId + '/events';
   var eventweek = '/v1/tags/' + this.tagId + '/eventWeek';
+  var query_timezone = moment.tz.guess();
   $scope.tagDetails = [];
   $scope.tagEvents = [];
+  $scope.tagHours = this.tagHours;
 
   $http({method: 'GET', url: tagUrl + '.json' }).
   success(function successCallback(data) {
     for (var i = 0; i < data.results.length; i++) {
       var event = data.results[i]
       $scope.tagEvents.push({
-        start: event.start,
+        start: (new Date(event.start)).toString(),
         name: event.name,
       });
     }
   });
-
-  $http({method: 'GET', url: eventweek + '.json' }).
+  $http({method: 'GET',
+         url: eventweek + '.json',
+         params: {
+           timezone: query_timezone,
+         }
+       }).
   success(function successCallback(data) {
+    $scope.tagHours = $scope.tagHours/data.length;
     var events = [];
+    var max_hour = 0;
     for (var i = 0; i < data.length; i++) {
```

Comment:
Some explanatory comments around this area would be particularly helpful, since it seems to be somewhat math/logic heavy.